mftf/2.3/versioning.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Versioning Policy for MFTF
+---
+
+This documemt describes the versioning policy for the Magento Functional Testing Framework (MFTF), including the version numbering schema.
+
+## Backwards Compatibility Definition
+
+Backwards compatibility is when a test undergoes changes, but achieves the same results as before and remains compatible with potential test customizations.
+
+Different types of changes that are backwards compatible:
+
+- **Test Flow change (Test/ActionGroup)** - A backwards compatible modification of a test flow would not diminish the original set of actions in the test.
+Some modifications may change an action's sequence or behavior, but still allows an extension to achieve the same test results without changing the test extension.
+- **Test Entity change (Data/Section/Page/Metadata)** - Adding new values or updating a `value` of existing entities which do **NOT** require updates to the test.
+- **Test Annotation change** - Annotations can be changed and are considered backward compatible changes.
+However, removing or changing a `<group />` annotation is considered a backward incompatible change.
+- Changes which delete and/or rename (Test/Action Group/Data/Metadata/Page/Section/Action)'s id attribute are considered as a backward incompatible change. Changing a reference to a data entity is considered a backward incompatible change.
+
+## Find your MFTF version number
+
+To find the current version of MFTF you are using, run the Magento CLI command:
+
+```bash
+cd <magento_root>/
+vendor/bin/mftf --version
+```
+
+## Versioning Policy
+
+MFTF versioning policy follows guidelines from [Semantic Versioning](https://semver.org/).
+
+### 3-component version numbers
+
+    X.Y.Z
+    | | |
+    | | +-- Backward Compatible changes (Patch release - bug fixes)
+    | +---- Backward Compatible changes (Minor release - new tests)
+    +------ Backward Incompatible changes (Major release)
+
+### Z release - patch
+
+  Patch version **Z** MUST be incremented if only backward compatible changes to tests are introduced.
+  For example, a fix which aims to resolve test flakiness might:
+
+- Update an unreliable selector.
+- Add a wait for an element.
+- Update a data entity value.
+  
+### Y release - minor
+
+  Minor version **Y** MUST be incremented if a new, backwards compatible test or test entity is introduced.
+  It MUST be incremented if any test or test entity is marked as deprecated.
+  It MAY include patch level changes. Patch version MUST be reset to 0 when minor version is incremented.
+
+### X release - major
+
+  Major version **X** MUST be incremented if any backwards incompatible changes are introduced to a test or test entity.
+  It MAY include minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.
```

Comment:
```suggestion
  It can include minor and patch level changes. You must reset the patch and minor version to 0 when you change the major version.
```